04-onion-routing.md

```diff
@@ -128,6 +128,53 @@ With the `hopfiller` being constructed in the same way as the routing
 info `filler` and each payload being incrementally obfuscated at each
 hop.
 
+### Per Hop Payload Format
+
+Using the per-hop-payload, the sender is able precisely specify the path and
+structure of the HTLC's forwarded at each hop. As the per-hop-payload is
+protected under the packet-wide HMAC, the information within the
+per-hop-payload is fully authenticated with each pair-wise relationship between
+the HTLC sender, and each intermediate node in the path.  Using this end-to-end
+authentication, forwarding nodes are able to ensure that the incoming node
+didn't forward an ill-crafted HTLC by cross-checking the HTLC parameters with
+the values as specified within the per-hop-payload.
+
+The format of the per-hop-payload for a version 0 packet is as follows: 
+```
+┌──────────────┬─────────────────────────┬───────────────────────────────┬───────────────────────────────────────────┐
+│realm (1 byte)│amt_to_forward (8 bytes) │ incoming_cltv_value (2 bytes) │unused_with_v0_version_on_header (9 bytes) │
+└──────────────┴─────────────────────────┴───────────────────────────────┴───────────────────────────────────────────┘
```

Comment:
Good point, I've modified the ASCII diagram to be consistent with the style used within the rest of the document.